src/nvim/os/fs.c

```diff
@@ -279,13 +279,24 @@ bool os_file_is_readonly(const char *name)
   return access(name, W_OK) != 0;
 }
 
+/// Check if a file is readable.
+///
+/// @return 0 if `name` is not writable,
+/// @return 1 if `name` is writable,
+/// @return 2 for a directory which we have rights to write into.
+bool os_file_is_readable(const char *name)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  return (access(name, R_OK) == 0);
```

Comment:
From that doc, it appears I may need to check 04 (Read-only) *and* 06 (Read and write). There is also a confusing note:

> This function only checks whether the file and directory are read-only or not, it does not check the filesystem security settings. For that you need an access token. 

I guess we'll find out if it works, once the tests are running. @equalsraf The unit tests (`make unittest`) do not require the lua-client; have you had any luck running them with MSVC?